packages/relay/src/lib/clients/sdkClient.ts

```diff
@@ -151,6 +162,13 @@ export class SDKClient {
     this.hbarLimitService = hbarLimitService;
     this.maxChunks = Number(ConfigService.get('FILE_APPEND_MAX_CHUNKS')) || 20;
     this.fileAppendChunkSize = Number(ConfigService.get('FILE_APPEND_CHUNK_SIZE')) || 5120;
+    this.mirrorNodeClient = new MirrorNodeClient(
```

Comment:
Ahh the classic dilemma of overlapping the mirror node and SDK client! This does go against our design of keeping these two separate. I’m sorry but I'd have to kindly object this implementation. Let's discuss tho! 

It looks like the mirror node was added here to poll for transactions if they time out. Since `sdkClient.submitEthereumTransaction()` already throws errors if there’s an issue with the transaction, and with the previous implementation (without this [try-catch](https://github.com/hashgraph/hedera-json-rpc-relay/pull/3161/files#diff-4b0bcc39d9efb3e923f041c7459f1cc42d34597d4200839df6764c2a345a1c1bR462-R484)), errors would eventually be captured in `EthImpl.sendRawTransaction()` at this [catch block here](https://github.com/hashgraph/hedera-json-rpc-relay/blob/main/packages/relay/src/lib/eth.ts#L1614-L1623). Since the error would remain consistent across all levels, how about handling the `isFailedTransaction` logic in that block instead?

Handling this in the `EthImpl` class would be beneficial, as there’s already a `mirrorNodeClient` instance available, which helps us avoid the dilemma of overlapping the two clients.

Reply:
Two things on this concern:
1. This is supposed to be a workaround, so its a temporary fix. I was also reluctant to adding mirrorNodeClient instance here, however it is inevitable because of #2 below
2. In the submitEthereumTransaction we have the following
`return {
      fileId,
      txResponse: await this.executeTransaction(
        ethereumTransaction,
        callerName,
        interactingEntity,
        requestDetails,
        true,
        originalCallerAddress,
      ),
    };`
and as you can see if we fail on this.executeTransaction(...) as it happens now, fileId wont'be returned and in this edge case we actually need it. If the transaction is found in the MN we will get the transactionId, but we need the fileId in order to proceed

Reply:
**Note:** Synced offline and agreed this is a temporary workaround. I still wanted to document my suggestion here for reference.

Still thinking up a solution that makes sense and avoids using the `mirror node client` within the `sdkClient`. It seems that when `executeTransaction()` throws an error, we're primarily concerned with the condition `e instanceof SDKClientError && (e.isConnectionDropped() || e.isTimeoutExceeded())`. Given this, [this block](https://github.com/hashgraph/hedera-json-rpc-relay/pull/3161/files#diff-4b0bcc39d9efb3e923f041c7459f1cc42d34597d4200839df6764c2a345a1c1bR473-R482) could be modified so the `transactionId` could be directly returned to `eth.sendRawTransaction()` without involving mirror node polling.

```diff
      if (e instanceof SDKClientError && (e.isConnectionDropped() || e.isTimeoutExceeded())) {
-        const isFailed = await this.isFailedTransaction(requestDetails, e.transactionId);
-        if (isFailed) {
-          throw e;
-        }
+        // Commenting out the polling block
+        // const isFailed = await this.isFailedTransaction(requestDetails, e.transactionId);
+        // if (isFailed) {
+        //  throw e;
+        // }
+      // Focusing only on transactionId if the error is due to timeout or connection drop
        txResponse = { transactionId: e.transactionId };
      } else {
        throw e;
      }
    }
```

The `transactionId` would then be relayed to `EthImpl.sendRawTransaction()` and used at [this line](https://github.com/hashgraph/hedera-json-rpc-relay/blob/main/packages/relay/src/lib/eth.ts#L1586). Finally, the polling logic can be applied in [this block here](https://github.com/hashgraph/hedera-json-rpc-relay/blob/main/packages/relay/src/lib/eth.ts#L1594-L1599). This approach would allow us to handle timeout errors while reusing the existing polling logic in `sendRawTransaction`.

Note that this is a top-of-mind suggestion, and I haven't tested it. If you'd like to try it, please test and adjust the code as needed. 

Also, this is purely for reference. I understand the urgency and there’s no hard pressure to implement this immediately. However, if this approach might help streamline the work, please consider it.

Reply:
Will followup in another comment

Reply:
@quiet-node is right and I prefer his suggestion.
Let's follow up in a separate PR in this sprint to address that if possible.
If it requires a significant refactor we can explore in a different sprint.
@konstantinabl please note the ticket to address this here, thanks.

Reply:
@Nana-EC @quiet-node here is the issue https://github.com/hashgraph/hedera-json-rpc-relay/issues/3207